chain/chain-primitives/src/error.rs

```diff
@@ -70,9 +70,9 @@ pub struct Error {
 
 #[derive(Clone, Eq, PartialEq, Debug, Fail)]
 pub enum ErrorKind {
-    /// The block doesn't fit anywhere in our chain.
-    #[fail(display = "Block is unfit: {}", _0)]
-    Unfit(String),
+    /// The block is already known
+    #[fail(display = "Block is known: {:?}", _0)]
+    BlockKnown(BlockKnownError),
```

Comment:
That's a good point. For context, `process_block` returns Error in many cases that is not problematic in the sense that the block is invalid or something wrong occurred during the process. It simply means that the block was not processed successfully and the reason why it wasn't processed successfully is encoded in the error types. For example, the block is already known, or the block is an orphan (we don't know the previous block yet), or the block is invalid, all these cases correspond to some chain error type. And the caller of `process_block` will process these errors accordingly. It is definitely true that such info can be returned as value instead of as error, it is probably better that way too.